tests/utils/test_decorators.py

```diff
@@ -0,0 +1,156 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import ast
+from typing import TYPE_CHECKING, Any
+
+import pytest
+
+from airflow.decorators import task
+
+if TYPE_CHECKING:
+    from airflow.decorators.base import Task
+DECORATORS = sorted(set(x for x in dir(task) if not x.startswith("_")) - {"skip_if", "run_if"})
+
+
+class TestDecoratorSource:
+    @staticmethod
+    def update_custom_operator_name(operator: Any, custom_operator_name: str):
+        custom_operator_name = (
+            custom_operator_name if custom_operator_name.startswith("@") else f"@{custom_operator_name}"
+        )
+        operator.__dict__["custom_operator_name"] = custom_operator_name
+
+    @staticmethod
+    def init_decorator(decorator_name: str):
+        decorator_factory = getattr(task, decorator_name)
+
+        kwargs = {}
+        if "external" in decorator_name:
+            kwargs["python"] = "python3"
+        return decorator_factory(**kwargs)
+
+    @classmethod
+    def parse_python_source(cls, task: Task, custom_operator_name: str | None = None) -> str:
+        operator = task().operator
+        if not hasattr(operator, "get_python_source"):
+            pytest.skip(f"Operator {operator} does not have get_python_source method")
+        if custom_operator_name:
+            cls.update_custom_operator_name(operator, custom_operator_name)
+        return operator.get_python_source()
+
+    @classmethod
+    def parse_decorator_names(cls, source: Task | str) -> list[str]:
+        if not isinstance(source, str):
+            source = cls.parse_python_source(source)
+        node = ast.parse(source)
+        func: ast.FunctionDef = node.body[0]  # type: ignore[assignment]
+        decorators: list[ast.Name] = func.decorator_list  # type: ignore[assignment]
+        return [decorator.id for decorator in decorators]
```

Comment:
These should be implemented with `@pytest.fixture` instead.

Reply:
I defined `decorator` as `fixture` as you said.
However, I don't know the right way to remove `parse_python_source`.